File: java/PE040.java
public class PE040
{
    public static void main(String[] args)
    {
        int[] a = new int[1000000+100];
        int[] b = new int[100];
        int i = 0;
        int j = 1;
        while(i < 1000000) {
            int k = j;
            j ++;
            int m = 0;
            while(k != 0) {
                b[m] = k%10;
                k /= 10;
                m ++;
            }
            m --;
            while(m >= 0) {
                a[i] = b[m];
                i++;
                m--;
            }
        }
        int res = a[0]*a[10-1]*a[100-1]*a[1000-1]*a[10000-1]*a[100000-1]*a[1000000-1];
        System.out.println(res);
    }
}

File: c/PE040.c
#include <stdio.h>
#define MAX 1000000
int a[MAX+100],b[100];
int main() {
    int i, j, k, m;
    k = 1;
    for(i = 1; ; i++){
        j = i;
        m = 0;
        while(j){
            b[m] = j%10;
            j /= 10;
            m++;
        }
        for(m--; m >= 0; m--){
            a[k] = b[m];
            k++;
        }
        if(k > MAX)
            break;
    }
    int res = a[1]*a[10]*a[100] * a[1000] * a[10000] * a[100000] * a[1000000];
    printf("%d\n", res);
    return 0;
}
